test/orders.js

```diff
@@ -1,28 +1,88 @@
 process.env.NODE_ENV = 'test';
 
 const { expect, should } = require('chai');
-const { topBuys } = require('./mockMethods');
-const { Buy, Sell, Pair } = require('../src/db');
-
+import { Buy, Sell, Pair, Position } from '../src/db';
+import { topBuys, topSells, processOrder, closeOrder } from '../src/db/methods';
 
 describe('hooks', function() {
   
-  beforeEach(function() {
-    //Erase any existing DB and initialize the schema models
-    Buy.sync({ force: true });
-    Sell.sync({ force: true });
-    Pair.sync({ force: true });
+  beforeEach(function(done) {
+    Buy.findAll()
+      .then(res => res.forEach(row => row.destroy()))
+      .then(() => Sell.findAll())
+      .then(res => res.forEach(row => row.destroy()))
+      .then(() => Position.findAll())
+      .then(res => res.forEach(row => row.destroy()))
+      .then(() => {
+        let orders = [];
+        for (let i = 1; i <= 10; i++) {
+          orders.push({userId: i, volume: 1, price: 10});
+        }
+        return Buy
+          .bulkCreate(orders);
+      })
+      // .then(() => topBuys(console.log))
+      // .then(res => console.log(res))
+      .then(() => {
+        let orders = [];
+        for (let i = 1; i <= 10; i++) {
+          orders.push({userId: i, volume: 1, price: 10});
+        }
+        return Sell
+          .bulkCreate(orders);
+      })
+      .then(() => done());
   });
 
   describe('orders', function() {
+    // processOrder({order: {userId: 1, volume: 1, price: 1.0}, type: 'BUY'});
     describe('topBuys function', function() {
       it('should correctly return the first 10 BUY orders', function(done) {
-        let result = [];
-        topBuys(top => result = [...top], Buy)
-          .then(() => {
-            expect(result.length).to.equal(10);
-            done();
+        topBuys(top => expect(top.length).to.equal(10))
+          .then(() => done())
+          .catch(err => done(err));
```

Comment:
Check out this article about how to test promises with mocha.

https://wietse.loves.engineering/testing-promises-with-mocha-90df8b7d2e35

Chai-as-promised is a good supplement extension to test promise

https://github.com/domenic/chai-as-promised